src/core/namespace.c

```diff
@@ -317,12 +328,16 @@ static int apply_mount(
         assert(what);
 
         r = mount(what, m->path, NULL, MS_BIND|MS_REC, NULL);
-        if (r >= 0)
+        if (r >= 0) {
                 log_debug("Successfully mounted %s to %s", what, m->path);
-        else if (m->ignore && errno == ENOENT)
-                return 0;
+        }
+        else {
+                if (m->ignore && errno == ENOENT)
+                        return 0;
+                log_debug("Failed mounting %s to %s: %s", what, m->path, strerror(errno));
+        }
 
-        return r;
+        return -errno;
```

Comment:
We should return `-errno` on error. This code returns `-errno` always. Should be something like this:
```
if (r >= 0) {
       ....
       return r;
} else {
       if (m->ignore && errno == ENOENT)
                return 0;
       return -errno;
}
```